sklearn/decomposition/tests/test_pca.py

```diff
@@ -135,6 +136,43 @@ def test_pca_randomized_solver():
                      svd_solver='randomized', random_state=0).svd_solver)
 
 
+def test_pca_lobpcg_solver():
```

Comment:
if this is essentially identical to `test_pca_arpack_solver` and `test_pca_randomized_solver`, can we just use a parametrized test to check all three, rather than copy-pasting?

Reply:
You are correct. I was just afraid to break already existing  `test_pca_arpack_solver` and `test_pca_randomized_solver` so went with a poor coder solution of  copy-pasting.

Reply:
for https://github.com/scikit-learn/scikit-learn/pull/12319#discussion_r275687052 the new lobpcg code internally substitutes tol = .0 with a "reasonable" default, to my recollection, so this comment can be probably be removed? I am unsure how ARPACK deals with  tol = .0